src/util/parse_options.cpp

```diff
@@ -82,3 +84,19 @@ banner_string(const std::string &front_end, const std::string &version)
   return "* *" + std::string(left_padding, ' ') + version_str +
          std::string(right_padding, ' ') + "* *";
 }
+
+int parse_optionst::main()
+{
+  // catch all exceptions here so that this code is not duplicated
+  // for each tool
+  try
+  {
+    return parse_options_baset::main();
+  }
+  catch(invalid_user_input_exceptiont &e)
+  {
+    std::cerr << e.what() << "\n";
```

Comment:
No. This must go into a message stream.

Reply:
It was like this in the original design, but then I saw during rebasing on develop, that recently a similar change was eliminated, so I thought that it must have been that this is more appropriate. If I am also to do that now, I will need to add a `messaget` into `parse_options_baset`, which will require changes in various places

Reply:
At the risk of contradicting @peterschrammel: I don't think a message stream can be used here, because setting up the desired one requires successful parsing of command-line options (picking, e.g., JSON output with a verbosity of 5). Hence I'm with @NlightNFotis that using `std::cerr` is correct here.